crates/client/rpc-core/src/utils.rs

```diff
@@ -211,7 +210,7 @@ fn _to_btree_map_entrypoints(
 }
 
 /// Returns the current Starknet block from the block header's digest
-pub fn get_block_by_block_hash<B, C, BE>(client: Arc<C>, block_hash: <B as BlockT>::Hash) -> Option<StarknetBlock>
+pub fn get_block_by_block_hash<B, C, BE>(client: &C, block_hash: <B as BlockT>::Hash) -> Option<StarknetBlock>
```

Comment:
I think you can get away with much less constraints:

```rust

pub fn get_block_by_block_hash<B, C, BE>(client: &C, block_hash: <B as BlockT>::Hash) -> Option<StarknetBlock>
where
    B: BlockT,
    C: HeaderBackend<B>,
{
    let header = client.header(block_hash).ok().flatten()?;
    let digest = header.digest();
    let block = find_starknet_block(digest).ok()?;
    Some(block)
}
```

It will also lift the need for the `'static` everywhere